src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\Console\Helper\Table;
+
+/**
+ * Encode a user's password.
+ *
+ * @author Sarah Khalil <mkhalil.sarah@gmail.com>
+ */
+class UserPasswordEncoderCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('security:encode-password')
+            ->setDescription('Encode a password.')
+            ->addArgument('password', InputArgument::OPTIONAL, 'Enter a password')
+            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.')
+            ->addArgument('salt', InputArgument::OPTIONAL, 'Enter the salt you want to use to encode your password.')
+            ->setHelp(<<<EOF
+
+The <info>%command.name%</info> command allows to encode a password using encoders
+that are configured in the application configuration file, under the <comment>security.encoders</comment>.
+
+For instance, if you have the following configuration for your application:
+<comment>
+    security:
+        encoders:
+            Symfony\Component\Security\Core\User\User: plaintext
+            AppBundle\Model\User: bcrypt
+</comment>
+
+According to the response you will give to the question "<question>Provide your configured user class</question>" your
+password will be encoded the way it was configured.
+    - If you answer "<comment>Symfony\Component\Security\Core\User\User</comment>", the password provided will be encoded
+      with the <comment>plaintext</comment> encoder.
+    - If you answer <comment>AppBundle\Model\User</comment>, the password provided will be encoded
+      with the <comment>bcrypt</comment> encoder.
+
+The command allows you to provide your own <comment>salt</comment>. If you don't provide any,
+the command will take about that for you.
+
+You can also use the non interactive
```

Comment:
Looks unfinished to me. While you are at it, can you rework the documentation to describe the non-interactive way first?